sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java

```diff
@@ -273,27 +273,30 @@ private void flush() {
     }
 
     private void exportBatches(ArrayList<ReadableSpan> spanList) {
-      ArrayList<Span> spansProtoList = new ArrayList<>(maxExportBatchSize);
       // TODO: Record a counter for pushed spans.
       for (int i = 0; i < spanList.size(); ) {
         int batchSizeLimit = Math.min(i + maxExportBatchSize, spanList.size());
-        for (int j = i; j < batchSizeLimit; j++) {
-          spansProtoList.add(spanList.get(j).toSpanProto());
-          // Remove the reference to the RecordEventsSpanImpl to allow GC to free the memory.
-          spanList.set(j, null);
-        }
+        List<SpanData> spanDataForExport = createSpanDataForExport(spanList, i, batchSizeLimit);
         // One full batch, export it now. Wrap the list with unmodifiableList to ensure exporter
         // does not change the list.
-        onBatchExport(Collections.unmodifiableList(spansProtoList));
-        // Cannot clear because the exporter may still have a reference to this list (e.g. async
-        // scheduled work), so just create a new list.
-        spansProtoList = new ArrayList<>(maxExportBatchSize);
+        onBatchExport(Collections.unmodifiableList(spanDataForExport));
         i = batchSizeLimit;
       }
     }
 
+    private List<SpanData> createSpanDataForExport(
+        List<ReadableSpan> spanList, int startIndex, int numberToTake) {
+      List<SpanData> spanDataBuffer = new ArrayList<>(maxExportBatchSize);
```

Comment:
This will always be numberToTake size, so please update.

Reply:
you are correct!

Reply:
done!